pkg/framework/webhook.go

```diff
@@ -0,0 +1,84 @@
+package framework
+
+import (
+	"os"
+
+	"github.com/averageflow/gohooks/v2/gohooks"
+	"gopkg.in/yaml.v2"
+	"k8s.io/klog/v2"
+
+	"path/filepath"
+)
+
+func SendWebhook(webhookConfig string) {
+	cfg, err := LoadConfig(webhookConfig)
+	if err != nil {
+		klog.Fatal(err)
+	}
+	path, err := os.Executable()
+	if err != nil {
+		klog.Info(err)
+	}
+
+	//Create webhook
+	hook := &gohooks.GoHook{}
+	w := Webhook{Path: path}
+	w.RepositoryURL = cfg.WebhookConfig.RepositoryURL
+	w.Repository.FullName = cfg.WebhookConfig.Repository.FullName
+	saltSecret := cfg.WebhookConfig.SaltSecret
+	hook.Create(w, path, saltSecret)
+
+	//Send webhook
+	resp, err := hook.Send(cfg.WebhookConfig.WebhookTarget)
+	if err != nil {
+		klog.Fatal("Error sending webhook: ", err)
+	}
+	klog.Info("Webhook response: ", resp)
+}
+
+// LoadConfig returns a decoded Config struct
+func LoadConfig(configPath string) (*Config, error) {
+	// Create config structure
+	config := &Config{}
+
+	// Open config file
+	file, err := os.Open(filepath.Clean(configPath))
+	if err != nil {
+		return nil, err
+	}
+
+	// Init new YAML decode
+	d := yaml.NewDecoder(file)
+
+	// Start YAML decoding from file
+	if err := d.Decode(&config); err != nil {
+		return nil, err
+	}
+
+	if err := file.Close(); err != nil {
+		klog.Fatal("Error closing file: %s\n", err)
+	}
+
+	return config, nil
+}
+
+// Config struct for webhook config
+type Config struct {
```

Comment:
Try to separate the structs and define them at the beggining of file. Example: https://github.com/redhat-appstudio/application-service/blob/main/api/v1alpha1/application_types.go#L27-L59